src/porepy/models/solution_strategy.py

```diff
@@ -183,9 +183,7 @@ def __init__(self, params: Optional[dict] = None):
         is adjusted during simulation. See :meth:`before_nonlinear_loop`.
 
         """
-
-        self.nonlinear_solver_statistics = pp.SolverStatistics()
```

Comment:
When we remove it from the init, I believe, type checkers might not be able to figure out what type it is. Correct me if I'm wrong.

I suggest adding a type annotation for it on the class level in SolutionStrategy: `nonlinear_solver_statistics: pp.SolverStatistics` and a docstring below.

Reply:
As a class attribute, you mean?

Reply:
Yes